docs/eh_v3.md

```diff
@@ -0,0 +1,140 @@
+# EH_v3 integration document
+
+To start listening EHorizon updates `EHorizonObserver` should be registered with
+
+```kotlin
+mapboxNavigation.registerEHorizonObserver(eHorizonObserver)
+```
+
+It provides the next callbacks:
+
+```kotlin
+/**
+ * Electronic horizon listener. Callbacks are fired in the order specified.
```

Comment:
> Callbacks are fired in the order specified.

If we're documenting this, we should have a test for it.

Reply:
@LukasPaczos not sure that we can test it because we need to setup a real NN navigator object.
I think NN side is responsible for that and should cover it with tests. 
cc @etl